### DISCUSSION TITLE
automatic re-configuration after automower turns off?

### DISCUSSION THREAD

**Discussion by tuupos** [Thomas55555/husqvarna_automower#200] [Q&A] [answered]
I've set up husqvara_automower and it works like a champ (use the beta, I now don't have any connection losses again).

Question: My children stop the mower often manually because they are afraid the mower will run over our chickens (which jump to the side btw). after being in this "manually stopped" state, the mower turns off. After turning off I turn in manually on again (no other option) and every time the integrtion needs to be re-configured so I have to put in the API key, name password again.

Is it possible to store this somewhere else so i fetches automatically the right secrets so I dont have to put it in again like:

husvarna_automower:
  apikey: !secret husqvarna_apikey
  username: !secret husqvarna_username
  password: !secret husqvarna_password

thanks

**Comment by Thomas55555** [comment, marked as answer]
It should actually not happen, that stopping the mower leads to a re-configuration. Can you turn on the debugger, simulate the situation and post the logs here?
For the re-authentication I've implemented a Authentication with the Authorization code Grant. It's in the current beta version available. You can try this if you like.
You have to add following to your `configuration.yaml`
```
husqvarna_automower:
  client_id: !secret husqvarna_apikey
  client_secret: !secret husqvarna_client_secret
```
Don't mix up the `client_secret` with your `password`. You can find the `client_secret` here `https://developer.husqvarnagroup.cloud/apps` in your app.
Also add your home assistant URLs to the `Redirect URIs`. In the best case both URIs. (external and internal), so you are flexible.
![DeepinBildschirmfoto_Bereich auswählen_20220428194425](https://user-images.githubusercontent.com/59625598/165815612-e52ad1b1-1e4f-44eb-ac18-e10a5f2db293.png)
If you need to re-authenticate now. And select the Authorization Code Grant method. You will get redirected to Husqvarna and you have to enter username and password. You can save this in your browser. And than you don't have to type it in every time.
What do you think about it?
